### ISSUE TITLE
Control  of collection layout

### ISSUE THREAD

**Issue by imoutsatsos** [Pin0/openseadragon-justified-collection#1]
I have a requirement for displaying multiple images on the canvas and this plugin nicely fulfills this requirement. However, I'm getting unexpected behavior in the image layout as I progressively add more images to the collection.

Below are the settings for the collection mode and the layout of 3, 6 and 7 images respectively.

![image](https://cloud.githubusercontent.com/assets/3408226/24775022/30a26daa-1ae9-11e7-92ae-2c10a0f1df6c.png)

Unfortunately, I have sets of three related images that need to be arranged consistently to make sense for visualization, perhaps like this:
![image](https://cloud.githubusercontent.com/assets/3408226/24775148/9eecf492-1ae9-11e7-9b6b-edaffc9f4463.png)

Are there any settings that I should change to achieve the desired layout?

Thanks for any feedback!
Ioannis

**Comment by Pin0** [comment]
I don't know how dynamic your images are when concerning the width of those images. However, the collection rows are only used in the thirst row. If you have collectionRows=3 and the sixth image has a width that when added to the 4th and 5th has a larger row width than the combined first 3. It will move to the next row. That's how this plugin is intended. You can have control over the order by adding the images in the way you want. If your image array is 1,4,7,2,5,8,3,6,9 and the width of your images is the same then you should be ok I think. 

I'm closing this issue but if you have anything to add please feel free to do so.

**Comment by imoutsatsos** [comment]
Thank you, I now understand the plugin layout strategy better. 

In my case, all images are exactly the same width, so they would consistently occupy the same space in each row. However, I have now abandoned using collection mode due to these limitations. Instead I'm adding images to the canvas using a custom layout strategy where I can specify rows OR columns as taking 'fill priority' when the images are added.  

So for example, to achieve the blue layout above, I use a ' 3, Rows-First' layout where I fill FIRST three rows before filling the next column. In other cases, I feel columns first before moving to the next row. I found that this layout provide users with the flexibility needed to compare series of multi-spectra microscopy images. We need images from the same wavelength to always be on the same column or row. See an example here: https://www.youtube.com/watch?v=DKXA3mLNvW8

Perhaps this functionality can be implemented int the plugin in the future for additional flexibility. Thank you for your consideration.
